### PR TITLE
[PF-2014] Revert DATA_COLLECTION resource changes

### PR DESCRIPTION
**Pull request by melissachang**
We changed how we're implementing data collections. Instead a referenced resource TERRA_WORKSPACE resource, we now clone individual data collection resources into folders.

This basically reverts the PRs listed in PF-2014.

`terra resolve` still works:

```
~/terra-cli (mc/cleanup): terra resolve --name=1000-genomes-2
projects/terra-vdevel-buoyant-plum-9341/locations/us-central1-a/instances/melchang-20220728-193014
~/terra-cli (mc/cleanup): terra resolve
Missing required option: '--name=<name>'
Usage: terra resolve [--exclude-bucket-prefix] [--bq-path=<bqPathFormat>]
                     [--format=<format>] --name=<name> [--workspace=<id>]
Resolve a resource to its cloud id or path.
      --bq-path=<bqPathFormat>
                          [For BIG_QUERY_DATASET and BIG_QUERY_DATA_TABLE]
                            Cloud id format: FULL_PATH=[project id].[dataset
                            id].[table id if applicable], DATASET_ID_ONLY=
                            [dataset id], PROJECT_ID_ONLY=[project id], [For
                            BIG_QUERY_DATA_TABLE only] TABLE_ID_ONLY=[data
                            table id]
                            Default: FULL_PATH
      --exclude-bucket-prefix
                          [For GCS_BUCKET and GCS_OBJECT] Exclude the 'gs://'
                            prefix.
      --format=<format>   Set the format for printing command output: JSON,
                            TEXT. Defaults to the config format property.
                            Default: null
      --name=<name>       Name of the resource, scoped to the workspace. Only
                            use letters, numbers, dashes, and underscores.
      --workspace=<id>    Workspace id to use for this command only.
~/terra-cli (mc/cleanup): terra resolve --name=melchangbucketwithnounderscore
gs://melchangbucketwithnounderscore
```

### CODE REVIEW

File: src/main/java/bio/terra/cli/businessobject/Workspace.java [hunk 1]
```diff
@@ -429,12 +420,4 @@ public String getUserEmail() {
...
     return Collections.unmodifiableList(resources);
   }
-
-  public OffsetDateTime getCreatedDate() {
-    return createdDate;
-  }
-
-  public OffsetDateTime getLastUpdatedDate() {
-    return lastUpdatedDate;
-  }
```
Comment (zloery) [inline src/main/java/bio/terra/cli/businessobject/Workspace.java:439]: These seem relevant even for non data collection workspaces, is it worth keeping them?

**Review by zloery — APPROVED** [review]

File: src/main/java/bio/terra/cli/businessobject/Workspace.java [hunk 2]
```diff
@@ -57,9 +56,6 @@ public class Workspace {
   // list of resources (controlled & referenced)
   private List<Resource> resources;
 
-  private OffsetDateTime createdDate;
```
Comment (yuhuyoyo) [inline src/main/java/bio/terra/cli/businessobject/Workspace.java:60]: why is this removed?

**Review by yuhuyoyo — APPROVED** [review]
